documentation-site/components/yard/types.ts

```diff
@@ -1,41 +1,69 @@
 import {PropTypes} from './const';
+import {Action} from './const';
+
+export type TDispatch = (value: {type: Action; payload: any}) => void;
+
+export type TThemeDiff = {
+  themeValues: {[key: string]: string};
+  themeName: string;
+};
 
 export type TPropHook = {
   what: string;
   into: string;
-} | null;
+};
 
-export type TExtraImports = {
+export type TImportsConfig = {
   [key: string]: {
     named?: string[];
     default?: string;
   };
 };
 
+export type TError = {
+  where: string;
+  msg: string | null;
+};
+
 export type TYardProps = {
   componentName: string;
   minHeight: number;
-  extraImports?: TExtraImports;
   scopeConfig: {[key: string]: any};
   propsConfig: {[key: string]: TProp};
   themeConfig: string[];
+  importsConfig?: TImportsConfig;
 };
 
+export type TConfig = {
+  scopeConfig: {[key: string]: any};
+  propsConfig: {[key: string]: TProp};
+  themeConfig: string[];
+  importsConfig?: TImportsConfig;
+};
+
+export type TPropValue =
+  | undefined
+  | boolean
+  | string
+  | {
+      [key: string]: {
+        active: boolean;
+        style: string;
+      };
+    };
```

Comment:
It took me a moment to realize what this represents. Could the override definition be extracted to its own named type?